File: cpp/45a_1760C_Advantage.cpp
#include<bits/stdc++.h>

int main(){
    int t, n, temp;
    std::vector<std::vector<int>> ans;
    std::cin>>t;
    for (int i = 0; i < t; i++){
        std::cin>>n;
        std::vector<int> arr, arr_temp, nxt;
        for(int j = 0; j < n; j++){
            std::cin>>temp;
            arr.push_back(temp);
            arr_temp.push_back(temp);
        }
        sort(arr_temp.begin(), arr_temp.end());
        for(auto j : arr){
            if (j == arr_temp[n-1]){
                nxt.push_back(j - arr_temp[n-2]);
                //std::cout<<"top => "<<nxt.back()<<"\n";
            }
            else{
                nxt.push_back(j - arr_temp[n-1]);
                //std::cout<<nxt.back()<<"\n";
            }
        }
        ans.push_back(nxt);
    }
    for(auto i : ans){
        for(auto j : i){
            std::cout<<j<<" ";
        }
        std::cout<<"\n";
    }
    return 0;
}

File: python/45a_1760C_Advantage.py
def main():
    ans = []
    t = int(input())
    for _ in range(t):
        n = int(input())
        arr = input().split()
        arr_temp = []
        for i in range(n):
            arr[i] = int(arr[i])
            arr_temp.append(arr[i])
        arr_temp.sort()
        nxt = []
        for i in arr:
            if i == arr_temp[-1]:
                nxt.append(i - arr_temp[-2])
            else:
                nxt.append(i - arr_temp[-1])
        ans.append(nxt)
    for i in ans:
        for j in i:
            print(j, end = " ")
        print()

if __name__ == "__main__":
    main()